django/contrib/admin/options.py

```diff
@@ -109,9 +109,8 @@ class BaseModelAdmin(six.with_metaclass(forms.MediaDefiningClass)):
     show_full_result_count = True
     checks_class = BaseModelAdminChecks
 
-    @classmethod
-    def check(cls, model, **kwargs):
-        return cls.checks_class().check(cls, model, **kwargs)
+    def check(obj, model, **kwargs):
+        return obj.checks_class().check(obj, model, **kwargs)
```

Comment:
Why? If this was a classmethod on InlineModelAdmin, it would have the same
problem as for ModelAdmin, wouldn't it?

On 10 September 2015 at 16:10, Simon Charette <notifications@github.com>
wrote:

> In django/contrib/admin/options.py
> <https://github.com/django/django/pull/5256#discussion_r39171326>:
>
> > @@ -109,9 +109,8 @@ class BaseModelAdmin(six.with_metaclass(forms.MediaDefiningClass)):
> >      show_full_result_count = True
> >      checks_class = BaseModelAdminChecks
> >
> > -    @classmethod
> > -    def check(cls, model, **kwargs):
> > -        return cls.checks_class().check(cls, model, **kwargs)
> > +    def check(obj, model, **kwargs):
> > +        return obj.checks_class().check(obj, model, **kwargs)
>
> Actually this should be classmethod for InlineModelAdmin and a regular
> method for ModelAdmin.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/django/django/pull/5256/files#r39171326>.
>